### ISSUE TITLE
New hard crash after targeting .net 9 from 8 (System.AccessViolationException)

### ISSUE THREAD

**Issue by bettwedder** [dotnet/runtime#110543]
### Description

With .NET 9 I started getting System.AccessViolationException crashes in a .NET MAUI application that was rock solid before the upgrade. The issue is inside a function that reads and sums arrays of longs in parallel using Enumberable.Sum though I am not able to reproduce the issue in debug mode yet.  All crashes have occurred on the release build.  The program runs the function that throws this error twice each millisecond and will run for anywhere from 25 minutes to 2 hours without it encountering the error. An example message is at the bottom of this post because while writing this post I got a new crash reason:

Application: MyMauiApp.exe
CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an internal error in the .NET Runtime at IP 0x00007FFE80BBEA8B (0x00007FFE808D0000) with exit code 0xc0000096.

Previous Crash Errors:

CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

Stack:
at System.Runtime.ExceptionServices.InternalCalls.g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
at System.Runtime.ExceptionServices.InternalCalls.g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
at System.Runtime.EH.RhThrowHwEx(UInt32, ExInfo ByRef)
at MyNamespace.MyClass+<>c__DisplayClass96_0.b__0(Int32)
at System.Threading.Tasks.Parallel+<>c__DisplayClass19_0`2[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Int32, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].b__1(System.Threading.Tasks.RangeWorker ByRef, Int64, Boolean ByRef)
at System.Threading.Tasks.TaskReplicator+Replica.Execute()
at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread)
at System.Threading.ThreadPoolWorkQueue.Dispatch()
at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()

### Reproduction Steps

I do not have the minimum steps at this time. 

### Expected behavior

Keep running and throw a catchable exception inside the app that my default handlers could process.

### Actual behavior

Program closes and writes an event to EventViewer

### Regression?

Did not occur in .NET 8

**Comment by mangod9** [comment]
are you able to share a repro and/or dumps for the AV?

**Comment by jkotas** [comment]
Likely duplicate of #108763

**Comment by bettwedder** [comment]
Sorry I don't have a repo for you, I am currently trying to figure out how to reproduce it but it is so intermittent.  I am going to try the remedy in #110088 of setting CETCompat to false.

In case hardware is suspected, this is running on AMD 7950X with Windows 11 22H2

**Comment by karmeli87** [comment]
Hi, we are very seeing similar failures in our tests after upgrading form dotnet 8 to 9
```
Application: testhost.exe
CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an unhandled exception.
Stack:
   at System.Runtime.ExceptionServices.InternalCalls.RhpCallFinallyFunclet(Byte*, Void*, ExInfo*)
   at System.Runtime.ExceptionServices.InternalCalls.RhpCallFinallyFunclet(Byte*, Void*, ExInfo*)
   at System.Runtime.EH.InvokeSecondPass(ExInfo ByRef, UInt32, UInt32)
   at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
   at System.Runtime.EH.RhRethrow(ExInfo ByRef, ExInfo ByRef)
   at Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib, 
Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, 
Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].ReceiveReplicationBatches()
   at Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib, 
Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, 
Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].DoIncomingReplication()
   at Raven.Server.Utils.PoolOfThreads+PooledThread.DoWork()
   at Raven.Server.Utils.PoolOfThreads+PooledThread.Run()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
```

We don't have a way to repro yet, but we do have a dump to share.
To where can we upload it securely?

**Comment by jkotas** [comment]
> We don't have a way to repro yet, but we do have a dump to share. To where can we upload it securely?

You can create an issue on https://developercommunity.visualstudio.com/, add the dump to it as a private attachment, and share the link here.

**Comment by jkotas** [comment]
> Hi, we are seeing very similar failures in our tests after upgrading form dotnet 8 to 9

This looks like a different failure. The crash tracked by this issue is System.AccessViolationException. The error message that you have shared does not have AccessViolationException and the stacktrace is different. You may want to create a new github issue to track it.

**Comment by mangod9** [comment]
@karmeli87 yeah please open a separate issue for the failure you are observing, that looks related to new Exception Handling which was enabled by default in 9.

**Comment by stephenweaver** [comment]
Also getting this  when updating from 8 to 9. I'll update to latest patch and try again.
```Application: StockScreener.Adhoc.exe
CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Stack:
   at System.Runtime.ExceptionServices.InternalCalls.<RhpSfiInit>g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
   at System.Runtime.ExceptionServices.InternalCalls.<RhpSfiInit>g____PInvoke|0_0(System.Runtime.StackFrameIterator*, Void*, Int32, Boolean*)
   at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
   at System.Runtime.EH.RhThrowHwEx(UInt32, ExInfo ByRef)
   at System.Collections.Generic.HashSet`1[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].CopyTo(System.__Canon[], Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..ctor(System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at System.Linq.Enumerable+UnionIterator`1[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].ToList()```

**Comment by jkotas** [comment]
> .NET Version: 9.0.0

Could you please update to the latest .NET 9 servicing update (9.0.3) and check whether it fixed the issue?

 https://dotnet.microsoft.com/en-us/download/dotnet/9.0

**Comment by stephenweaver** [comment]
@jkotas It ran the rest of today without crashing, so I'd say this is fixed for me. thanks.

**Comment by jkotas** [comment]
@bettwedder Could you please check whether your crash was fixed in 9.0.3 as well? If not, could you please share the dump if possible (https://github.com/dotnet/runtime/issues/110543?reload=1#issuecomment-2545793077)?

**Comment by bettwedder** [comment]
I have tested live for a couple of days now and do not get the access violation anymore, but, I have the CETFlag set to false for my AMD 7950X for safety.
